widgetssdk/src/main/java/com/glia/widgets/launcher/EngagementLauncher.kt

```diff
@@ -2,50 +2,96 @@ package com.glia.widgets.launcher
 
 import android.content.Context
 import com.glia.androidsdk.Engagement
+import com.glia.widgets.callvisualizer.controller.CallVisualizerContract
 import com.glia.widgets.chat.Intention
 import com.glia.widgets.core.secureconversations.domain.HasOngoingSecureConversationUseCase
-import com.glia.widgets.di.ControllerFactory
 import com.glia.widgets.engagement.domain.EndEngagementUseCase
+import com.glia.widgets.engagement.domain.EngagementTypeUseCase
 import com.glia.widgets.engagement.domain.IsQueueingOrLiveEngagementUseCase
 import com.glia.widgets.helper.Logger
 import com.glia.widgets.helper.TAG
 
+@RequiresOptIn(
+    level = RequiresOptIn.Level.ERROR,
+    message = "This feature is under consideration for removal in a future releases. Use with caution."
+)
+@Retention(AnnotationRetention.BINARY)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY)
+/**
+ * Marks the feature as under consideration for removal in a future releases.
+ */
+annotation class FeatureUnderConsiderationForRemoval
```

Comment:
If we do decide to use this approach, I would suggest to move this class out of `EngagementLauncher`. It looks like a standalone class. Also, it doesn't look good to ship the brand new `EngagementLauncher` with anything non-recommended or even deprecated, I think.

Reply:
> If we do decide to use this approach, I would suggest to move this class out of EngagementLauncher. It looks like a standalone class. 

Agree!

Reply:
> Also, it doesn't look good to ship the brand new EngagementLauncher with anything non-recommended or even deprecated, I think.

But the `visitorContextAssetId` is actually non-recommended, isn't it?

Reply:
'Is a poorly-thought feature'. 🙂